[chore] Bump BE deps at the start of Metabase 54

bshepherdson opened the pull request:
Closes #52843.

### Description

Bump the BE versions (core and drivers) to the latest versions.

### How to verify

Tests passing, app still works.

### Checklist

- ~~Tests have been added/updated to cover changes in this PR~~

bshepherdson committed:
[chore] Bump BE deps at the start of Metabase 54

Fixes #52843.

bshepherdson commented:
* **#52853** <a href="https://app.graphite.dev/github/pr/metabase/metabase/52853?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/metabase/metabase/52853?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

metamben reviewed (commented):
LGTM, but redshift might have broken something.

bshepherdson commented:
> LGTM, but redshift might have broken something.

Yeah, for once the weird Redshift failures were legitimate. The Redshift drivers have an "optional" dep on `com.amazonaws/aws-java-sdk-core`, but apparently we're calling some path that now needs a `StringUtils` class from it, but it's not getting (transitively) depended on.

Adding an explicit dependency on that has got the Redshift tests passing locally, so I'll send that to CI.

bshepherdson committed:
Redshift: Adding an explicit dep on `aws-java-sdk-core`

This is listed as an optional dep, but it seems that at least one class
from it (`com.amazonaws.util.StringUtils`) is now dynamically required
by our driver - tests were throwing `NoClassDefFoundError`s without it.

Redshift tests are passing locally with this dependency added.

paoliniluis commented:
https://github.com/aws/amazon-redshift-jdbc-driver/issues/132

paoliniluis commented:
We should be able to remove the sdk dependency on their next release
